docs/guide/PINs/PIN-8-Listener-Flows.md

```diff
@@ -0,0 +1,126 @@
+- running a model with dedicated infrastructure (such as a cluster)
+
+## Proposal
+
+### New Requirements
+
+- A new `FlowRunner` subclass called `ListenerFlowRunner`
+- An `Event` class that has an arbitrary JSON-esque `value` and an optional `ack()` method for acknowledging the event
+- A `Listener` class whose `listen()` method is a generator that emits `Events`. It is attached to the `ListenerFlowRunner` as the `listener` attribute
+- The `ListenerFlowRunner` designates one of the flow's `Parameters` as the `event_parameter`. When an event is emitted by the `listener`, the flow will be executed with the event passed to that `event_parameter`.
+- A new `Listening` state that subclasses `Running`
+- A new schedule called `AlwaysRunning` which attempts to immediately restart a flow if it stops
+
+### Process
+
+This process bears some resemblance to Prefect's `mapping` semantic, where users define a process for handling a single item and Prefect automatically applies it to a list. Here, users build flows for processing a single event, and the `ListenerFlowRunner` automatically applies it to a stream.
+
+When `ListenerFlowRunner.run()` is called, the runner enters a `Listening` state rather than a `Running` state. It then begins looping over its `Listener.listen()` method, which is a generator of `Events`. Each `Event` has an `Event.value` attribute containing the actual value of the event (whatever that might be, including nothing) and an `Event.ack()` method allowing the event to be acknowledged, if appropriate. By default, this method does nothing.
+
+Each event is passed into a separate call of the `FlowRunner.run()` superclass method as the appropriately defined parameter value. Each of these "runs" represents a completely distinct processing of an event, and is therefore independently retry-able. The runner no longer defines success or failure from its tasks, but rather from whether it encounters an error in the listener.
+
+Each event therefore triggers a completely new flow run, but from a long-running process that replaces some of the latency of spinning up new architecture.
+
+Here is pseudocode for the process:
+
+```Python
+# inside ListenerFlowRunner
+def run(self, *args, parameters, context, **kwargs):
+
+    try:
+        # set state LISTENING
+        for event in self.listener.listen():
+
+            # add the event to parameters
+            run_params = parameters.copy()
+            run_params.update({self.event_parameter.name: event.value})
+
+            # --
+            # if running in Cloud, generate a new Submitted flow run at
+            # this time and pass the information to self.run
+            # self.client.create_flow_run(parameters=run_params)
+            # context.update(...)
+            # --
+
+            # acknowledge the event
+            event.ack()
+
+            # run the flow
+            super().run(*args, parameters, context=context, **kwargs)
```

Comment:
For my own clarity here: this couldn't happen as written because `super().run()` will be blocking.  Presumably (in Cloud at least), this would actually call something like `client.create_flow_run` and go on with its business. (Actually though, even that raises some infrastructure re-use questions for me). Could there be a local analogue?  We can discuss offline if the response is too in the weeds.

Reply:
Sorry, you actually made this comment on the issue and I believe I adjusted there, but didn't bring that along to the PR. I'll update now. In short, we just need to submit this line to the executor.